pkg/interface/chat/src/js/components/lib/ship-search.js

```diff
@@ -0,0 +1,339 @@
+      e.preventDefault();
+      e.stopPropagation();
+      this.props.onDismiss();
+    });
+  }
+
+  unbindShortcuts() {
+    if (!this.state.bound) {
+      return;
+    }
+    this.setState({ bound: false });
+    this.mousetrap.unbind("enter");
+    this.mousetrap.unbind("tab");
+    this.mousetrap.unbind(["up", "shift+tab"]);
+    this.mousetrap.unbind("down");
+    this.mousetrap.unbind("esc");
+  }
+
+  nextAutocompleteSuggestion(backward = false) {
+    const { suggestions } = this.state;
+    let idx = suggestions.findIndex(s => s === this.state.selected);
+
+    idx = backward ? idx - 1 : idx + 1;
+    idx = idx % suggestions.length;
+    if (idx < 0) {
+      idx = suggestions.length - 1;
+    }
+
+    this.setState({ selected: suggestions[idx] });
+  }
+
+  render() {
+    const { onSelect, contacts, popover, className } = this.props;
+    const { selected, suggestions } = this.state;
+
+    if (suggestions.length === 0) {
+      return null;
+    }
+
+    const popoverClasses = (popover && " absolute ") || " ";
+    return (
+      <div
+        style={
+          popover
+            ? {
+                bottom: "90%",
+                left: "48px"
+              }
+            : {}
+        }
+        className={
+          "black white-d bg-white bg-gray0-d " +
+            "w7 pv3 z-1 mt1 ba b--gray1-d b--gray4" +
+            popoverClasses +
+            className || ""
+        }
+      >
+        {suggestions.slice(0, 5).map(ship => (
+          <ShipSearchItem
+            onSelect={onSelect}
+            key={ship}
+            selected={selected}
+            contacts={contacts}
+            ship={ship}
+          />
+        ))}
+      </div>
+    );
+  }
+}
+
+export class ShipSearchInput extends Component {
+  constructor() {
+    super();
+    this.state = {
+      searchTerm: ""
+    };
+
+    this.inputRef = React.createRef();
+    this.popoverRef = React.createRef();
+
+    this.search = this.search.bind(this);
+
+    this.onClick = this.onClick.bind(this);
+  }
+
+  onClick(event) {
+    const { popoverRef } = this;
+    // Do nothing if clicking ref's element or descendent elements
+    if (!popoverRef.current || popoverRef.current.contains(event.target)) {
+      return;
+    }
+
+    this.props.onDismiss();
+  }
+
+  componentDidMount() {
+    document.addEventListener("mousedown", this.onClick);
+    document.addEventListener("touchstart", this.onClick);
+    this.inputRef.current.focus();
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener("mousedown", this.onClick);
+    document.removeEventListener("touchstart", this.onClick);
+  }
+
+  search(e) {
+    const searchTerm = e.target.value;
+    this.setState({ searchTerm });
+  }
+
+  render() {
+    const { state, props } = this;
+
+    return (
+      <div
+        ref={this.popoverRef}
+        style={{ top: "150%", left: "-80px" }}
+        className="b--gray2 b--solid ba absolute bg-white bg-gray0-d shadow-5"
```

Comment:
Idk, do we use drop shadows? @urcades It seems like removing the shadow and letting the border speak would be enough.

![image](https://user-images.githubusercontent.com/20846414/79285316-3a300180-7e8b-11ea-8793-96ed3bf90245.png)

Reply:
Yeah, let's null this drop shadow for now — there's no spec for their use in the latest Indigo works, and I think if we assemble our interfaces with clarity in mind in a general sense, we might not need them for a while.